tests/testthat/test-qenv_eval_code.R

```diff
@@ -105,6 +105,16 @@ testthat::test_that("eval_code accepts calls containing only comments and empty
   testthat::expect_identical(get_code(eval_code(qenv(), code)), code)
 })
 
+testthat::test_that("eval_code does not treat := as an assignment operator", {
+  testthat::skip_if_not_installed("data.table")
+  code <- "
+  iris <- data.table::data.table(iris) %>%
+    .[, NewSpecies := factor(Species)]
+  "
+  q <- eval_code(qenv(), code)
```

Comment:
we are running `eval_code` in this test. this test evaluates `code`. that's why we need `data.table` in Suggests.
We could change this test so it evaluates code from this comment, https://github.com/insightsengineering/teal.code/pull/234/files#r1860521594 , but this would test internals.

Another alternative is to test this in `teal.data` where we can test `teal_data(object, code  = code)` where `code` is not evaluated in this case. It is only parsed.

Reply:
changed to rlang https://github.com/insightsengineering/teal.code/pull/234/commits/6d38745e083fb668108850da3dbac92a18c3b747